code/datums/recipe.dm

```diff
@@ -94,6 +94,9 @@
 			O.reagents.trans_to(result_obj, O.reagents.total_volume)
 		if(!name_finalized && O.made_from_player)
 			result_obj.name = O.made_from_player + result_obj.name
+			if (istype(result_obj, /obj/item/reagent_container/food/snacks))
```

Comment:
again this is shit from OOP make a proc that runs on the result obj and has diff behaviour dont use shitty istype checks